opentelemetry-sdk/src/opentelemetry/sdk/logs/__init__.py

```diff
@@ -0,0 +1,174 @@
+        self.trace_id = trace_id
+        self.span_id = span_id
+        self.trace_flags = trace_flags
+        self.severity_text = severity_text
+        self.severity_number = severity_number
+        self.name = name
+        self.body = body
+        self.resource = resource
+        self.attributes = attributes
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, LogRecord):
+            return NotImplemented
+        return self.__dict__ == other.__dict__
+
+
+class LogData:
+    """Readable LogRecord data plus associated InstrumentationLibrary."""
+
+    def __init__(
+        self,
+        log_record: LogRecord,
+        instrumentation_info: InstrumentationInfo,
+    ):
+        self.log_record = log_record
+        self.instrumentation_info = instrumentation_info
+
+
+class LogProcessor(abc.ABC):
+    """Interface to hook the log record emitting action.
+
+    Log processors can be registered directly using
+    :func:`LogEmitterProvider.add_log_processor` and they are invoked
+    in the same order as they were registered.
+    """
+
+    @abc.abstractmethod
+    def emit(self, log_data: LogData):
+        """Emits the `LogData`"""
+
+    @abc.abstractmethod
+    def shutdown(self):
+        """Called when a :class:`opentelemetry.sdk.logs.LogEmitter` is shutdown"""
+
+    @abc.abstractmethod
+    def force_flush(self, timeout_millis: int = 30000):
+        """Export all the received logs to the configured Exporter that have not yet
+        been exported.
+
+        Args:
+            timeout_millis: The maximum amount of time to wait for logs to be
+                exported.
+
+        Returns:
+            False if the timeout is exceeded, True otherwise.
+        """
+
+
+class LogEmitter:
+    # TODO: Add multi_log_processor
+    def __init__(
+        self,
+        resource: Resource,
+        instrumentation_info: InstrumentationInfo,
+    ):
+        self._resource = resource
+        self._instrumentation_info = instrumentation_info
+
+    def emit(self, record: LogRecord):
+        # TODO: multi_log_processor.emit
+        pass
+
+    def flush(self):
+        # TODO: multi_log_processor.force_flush
+        pass
+
+
+class LogEmitterProvider:
+    # TODO: Add multi_log_processor
+    def __init__(
+        self,
+        resource: Resource = Resource.create(),
+        shutdown_on_exit: bool = True,
+    ):
+        self._resource = resource
+        self._at_exit_handler = None
+        if shutdown_on_exit:
+            self._at_exit_handler = atexit.register(self.shutdown)
+
+    def get_log_emitter(
+        self,
+        instrumenting_module_name: str,
+        instrumenting_module_verison: str = "",
+    ) -> LogEmitter:
+        return LogEmitter(
+            self._resource,
+            InstrumentationInfo(
+                instrumenting_module_name, instrumenting_module_verison
+            ),
+        )
+
+    def add_log_processor(self, log_processor: LogProcessor):
```

Comment:
Isn't this weird? I mean, this is a method of the `LogEmitterProvider` class

Reply:
I am not sure I understand you question. How else would we add a log processor?